crates/bevy_asset/src/server/mod.rs

```diff
@@ -1199,9 +1203,18 @@ pub enum LoadState {
     /// The asset has been loaded and has been added to the [`World`]
     Loaded,
     /// The asset failed to load.
-    Failed,
+    Failed(Box<AssetLoadError>),
 }
 
+impl PartialEq for LoadState {
```

Comment:
I don't think we want this: IMO we want to use the standard derived equality here. If users want this behavior it's easy for them to use it manually.

Reply:
Standard derived equality requires recursive equality implementations for all nested structures, should I add derived equality for all of them?

Reply:
Yes please :) Being able to check that the error matches the reason you expect is really valuable for writing tests, both internally and externally.